tests/helpers/mod.rs

```diff
@@ -0,0 +1,84 @@
+use std::fs::{self, remove_dir_all, remove_file, File};
+use std::io::Read;
+use std::net::TcpListener;
+use std::path::{Path, PathBuf};
+
+pub mod cmd;
+pub mod constants;
+pub mod setup;
+
+use cmd::exec;
+
+pub fn get_available_port() -> Option<u16> {
+	for port in 1025..65535 {
+		if TcpListener::bind(("127.0.0.1", port)).is_ok() {
+			return Some(port);
+		}
+	}
+
+	None
+}
+
+pub fn read_snapshot(log_dir: PathBuf, texts_to_hide: &[&str]) -> String {
+	let entry = log_dir.read_dir().unwrap().next().unwrap().unwrap();
+	let mut file = File::open(entry.path()).unwrap();
+	let mut buf = String::new();
+	file.read_to_string(&mut buf).unwrap();
+	for text_to_hide in texts_to_hide.iter() {
+		buf = buf.replace(text_to_hide, "{REDACTED}");
+	}
+	buf
+}
+
+pub fn clean_directory(dir: PathBuf) {
+	for f in dir.read_dir().unwrap() {
+		let f = f.unwrap();
+		let _ = if f.metadata().unwrap().is_dir() {
+			remove_dir_all(f.path())
+		} else {
+			remove_file(f.path())
+		};
+	}
+}
+
+pub fn initialize_repository(repo_dir: &Path, initial_branch: &str) {
+	exec::<&str, PathBuf>(
```

Comment:
See https://github.com/paritytech/parity-processbot/pull/338#discussion_r732314026

Reply:
I'm not opposed to using a library, but it definitely will not be done in this MR. Feel free to open a new issue for that.

Food for thought: even though using that library might subjectively appear cleaner, I infer it would increase complexity because, while we can reason about git commands which we use on a daily basis, by introducing that library all maintainers would have to learn about how it works in order to work with this code (which is merely a test helper, not actually application code). From my point-of-view the tradeoff is not worth it given that this `exec` interface which works well enough (including exit code, stderr visibility, etc...) for any commands we want to execute, not only git ones.